modules/widgets/blog-stats.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Blog Stats Widget.
+ *
+ * @since 4.5.0
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Disable direct access/execution to/of the widget code.
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Blog Stats Widget.
+ *
+ * Displays all time stats for that site.
+ *
+ * @since 4.5.0
+ */
+class Jetpack_Blog_Stats_Widget extends WP_Widget {
+
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		$widget_ops = array(
+			'classname' => 'blog-stats',
+			'description' => __( 'Show a hit counter for your blog.', 'jetpack' ),
+			'customize_selective_refresh' => true,
+		);
+		parent::__construct(
+			'blog-stats',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name', __( 'Blog Stats', 'jetpack' ) ),
+			$widget_ops
+		);
+		$this->alt_option_name = 'widget_statscounter';
+	}
+
+	/**
+	 * Return an associative array of default values
+	 *
+	 * These values are used in new widgets.
+	 *
+	 * @return array Array of default values for the Widget's options
+	 */
+	public function defaults() {
+		return array(
+			'title' => __( 'Blog Stats', 'jetpack' ),
+			'hits'  => __( 'hits', 'jetpack' ),
+		);
+	}
+
+	/**
+	 * Return All Time Stats for that blog.
+	 *
+	 * We query the WordPress.com Stats REST API endpoint.
+	 *
+	 * @uses stats_get_from_restapi(). That function caches data locally for 5 minutes.
+	 *
+	 * @return string|false $views All Time Stats for that blog.
+	 */
+	public function get_stats() {
+		// Get data from the WordPress.com Stats REST API endpoint.
+		$stats = stats_get_from_restapi( array( 'fields' => 'stats' ) );
+
+		if (
+			isset( $stats )
+			&& ! empty( $stats )
+			&& isset( $stats->stats->views )
```

Comment:
This could be simplified by checking only the last one, `isset( $stats->stats->views )`. This test shows that it's the same:
```php
var_dump( isset( $stats->stats->views ) ); // false

$stats = new stdClass();
var_dump( isset( $stats->stats->views ) ); // false
var_dump( isset( $stats ) && ! empty( $stats ) && isset( $stats->stats->views ) ); // false

$stats->stats = new stdClass();
var_dump( isset( $stats->stats->views ) ); // false
var_dump( isset( $stats ) && ! empty( $stats ) && isset( $stats->stats->views ) ); // false

$stats->stats->views = 'test';
var_dump( isset( $stats->stats->views ) ); // true
var_dump( isset( $stats ) && ! empty( $stats ) && isset( $stats->stats->views ) ); // true
```